src/libcore/ops.rs

```diff
@@ -795,41 +800,56 @@ not_impl! { bool usize u8 u16 u32 u64 isize i8 i16 i32 i64 }
 ///
 /// # Examples
 ///
+/// In this example, the `&` operator is lifted to a trivial `Scalar` type.
+///
+/// ```
+/// use std::ops::BitAnd;
+///
+/// #[derive(Debug, PartialEq)]
+/// struct Scalar(bool);
+///
+/// impl BitAnd for Scalar {
+///     type Output = Self;
+///
+///     fn bitand(self, Scalar(rhs): Self) -> Self {
+///         let Scalar(lhs) = self;
```

Comment:
Could you some comments here (at least to explain a bit about `lhs` and `rhs`, I think some people don't know what this is).